### ISSUE TITLE
REPL cursor should stop blinking when focus is elsewhere

### ISSUE THREAD

**Issue by byorgey** [swarm-game/swarm#776]
It also has been bothering me that the cursor continues to blink in the repl pane when focus is in another pane.

_Originally posted by @fryguybob in https://github.com/swarm-game/swarm/pull/151#discussion_r720825641_

**Comment by xsebek** [comment]
OK, I looked into this and it is not great. Basically, we abused a "form with one entry" to create the REPL and used the same `Name` type.

I tried using `setFormFocus` but it does not seem to work. My explanation is that the `FocusRing` of REPL one element form is just `REPLInput` and it will always be selected.

The whole dance around this is absurd, we do not need to store `Form`, the `REPLPrompt -> FormFieldState REPLPrompt AppEvent Name` for that one field is always the same. 🤦 The three copy-pasted lenses with confusing names updating REPL text should have alerted us that we are doing something wrong.

I think the problem is that several people (me included) added features to the REPL, but did not take the time to understand how the other features work and refactor it.
